.travis.yml

```diff
@@ -38,19 +38,19 @@ before_script:
 
 install:
   - sudo apt-get remove ipython
-  - pip install --upgrade pip
-  - pip install --upgrade setuptools
-  - pip install --upgrade wheel
+  - pip install --upgrade pip setuptools wheel
   - pip install $IPYTHON
+  # Install the desired version of numpy
   - pip install --only-binary=numpy,scipy numpy==$NUMPY_VERSION scipy
-  - pip install .[MongoTrials]
-  - pip install .[SparkTrials]
-  - pip install .[ATPE]
-  - pip install pytest>=3.6
-  - pip install pytest-cov pep8 pytest-pep8
-  - pip install wrapt
+  # Install the package itself with all extras
+  - pip install '.[MongoTrials, SparkTrials, ATPE]'
+  # Install test packages
+  - pip install pytest>=3.6 pytest-cov pep8 pytest-pep8 black
```

Comment:
Given that pip now has a real dependency resolver, would it be better to have even few pip commands with more dependencies in each command?

Also, Python 3.5 is end-of-life and Python 3.8 and 3.9 are not currently being tested.

Reply:
we're moving to python >= 3.7, see my other PR https://github.com/hyperopt/hyperopt/pull/742. I think 3.7 and 3.8 is enough for now, let's do 3.9 when it's a bit more mature

re pip: let's tackle that in a separate PR but I like the idea of having a setup.py and be it dependant on requirements.txt files. Or we could use something like piptools  https://github.com/jazzband/pip-tools

Reply:
regarding changes in this PR, I'm happy with something more compact, so LGTM